[wenjebs] iP

wenjebs opened the pull request:
# WenJigglyBot
> “Task management made simple.” – WenJigglyBot (source)

WenJigglyBot simplifies your task management. It's:

- text-based
- easy to use
- **FAST SUPER FAST** to get started

All you need to do is:

1. Clone the repository.
2. Run the `WenJigglyBot` application.
3. Start managing your tasks.

And it is **FREE**! 😉

## Features:
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Searching tasks
- [ ] Marking tasks as completed or uncompleted
- [ ] Deleting tasks

If you are a Java programmer, you can also use it to practice Java. Here's the main method:

```java
public static void main(String[] args) {
    new WenJigglyBot().run();
}

Commit:
Add Gradle support

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

wenjebs committed:
Rename bot, add greeting and exit

wenjebs committed:
Add save and displaying functionality

wenjebs committed:
Add marking and unmarking

wenjebs committed:
Add ToDo task support

wenjebs committed:
Add deadline task support

wenjebs committed:
Format text and support Event tasks

wenjebs committed:
Add testing

wenjebs committed:
Tweak tests

wenjebs committed:
Handle improper deadline input

wenjebs committed:
Add input processing, finalise error handling

wenjebs committed:
Add more tests for exception handling

wenjebs committed:
Add delete functionality

wenjebs committed:
Refactor to use enum

wenjebs committed:
Implement saving to file

wenjebs committed:
Add task initialisation on startup

wenjebs committed:
Make deadline task store LocalDate

wenjebs committed:
Fix local storage task parsing to use LocalDate

wenjebs committed:
Fix improper list for todo task

wenjebs committed:
Refactor Ui functions to a separate class

wenjebs committed:
Refactor storage components to separate class

wenjebs committed:
Refactor parsing functions to own class

wenjebs committed:
Refactor file saving functions to storage class

wenjebs committed:
Add classes to package, wenjigglybot

wenjebs committed:
Merge branch 'A-Gradle'

wenjebs committed:
Write unit test for Parser

wenjebs committed:
Write unit test for Parser

wenjebs committed:
Write unit tests for TaskList

wenjebs committed:
Write unit tests for TaskList

wenjebs committed:
Change gradle for jar building

wenjebs committed:
Change gradle for jar building

wenjebs committed:
Write Javadocs

wenjebs committed:
Write Javadocs for Parser, Storage, Task, Tasklist, Ui, Wenjigglybot

wenjebs committed:
Add find feature

wenjebs committed:
Add find feature

wenjebs committed:
Merge branch 'branch-Level-9'

wenjebs committed:
Merge branch 'branch-A-JavaDoc'

wenjebs committed:
Add docs to test

wenjebs committed:
Merge branch 'branch-A-CodingStandard'

wenjebs committed:
Setup checkstyle

starchypotatocode commented on build.gradle:
```diff
@@ -0,0 +1,46 @@
+}
+
+shadowJar {
+    archiveBaseName = "wenjigglybot"
```
Shouldn't this be capitalized for the .jar to be produced accordingly?

starchypotatocode commented on src/main/java/wenjigglybot/DeadlineException.java:
```diff
@@ -0,0 +1,9 @@
+package wenjigglybot;
+
+public class DeadlineException extends Exception {
```
These classes also need JavaDocs for the overall description

starchypotatocode commented on src/main/java/wenjigglybot/DeadlineTask.java:
```diff
@@ -0,0 +1,33 @@
+    private String deadlineString;
+    private LocalDate deadline;
+
+    public DeadlineTask(String description, LocalDate deadline) {
```
Likewise the constructors here and elsewhere

starchypotatocode commented on src/main/java/wenjigglybot/Parser.java:
```diff
@@ -0,0 +1,134 @@
+                    taskDescription = splitted[0];
+                }
+                String deadline = deadlineParts[1].replace(")", "").trim();
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d MMMM yyyy");
```
If you want, you could make this a private static final variable to make keeping things consistent easier?

wenjebs replied:
Sounds good!

starchypotatocode commented on src/main/java/wenjigglybot/TaskList.java:
```diff
@@ -0,0 +1,73 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskList {
```
Should there be an empty line between the two?

wenjebs committed:
Setup basic GUI skeleton

wenjebs committed:
Add list command functionality

wenjebs committed:
Add todo functionality with gui

wenjebs committed:
Add task marking in gui

wenjebs committed:
Finish up all gui features

wenjebs committed:
Fix storage load parse in bug

Jianwen0451 commented on src/main/java/wenjigglybot/Parser.java:
```diff
@@ -0,0 +1,138 @@
+     */
+    public static Task parseTask(String line) {
+        // Example format: 1. [T][X] task description
+        String[] parts = line.split(" ", 3);
```
Maybe change this name of parts to commandParts for more clarity?

wenjebs replied:
👍

Jianwen0451 commented on src/main/java/wenjigglybot/Parser.java:
```diff
@@ -0,0 +1,138 @@
+                } else {
+                    taskDescription = splitted[0];
+                }
+                String deadline = deadlineParts[1].replace(")", "").trim();
```
Maybe change this deadline to deadlineDate to prevent mixing up with the deadline task?

wenjebs replied:
Sure!

Jianwen0451 commented on src/main/java/wenjigglybot/Parser.java:
```diff
@@ -0,0 +1,138 @@
+     * @return A string array containing the event description, start time, and end time.
+     * @throws EventException If the input string does not contain the correct format.
+     */
+    public static String[] processEventTask(String task) throws EventException {
```
Maybe change the name task to input since the type is String rather than a Task?

Jianwen0451 commented on src/main/java/wenjigglybot/Task.java:
```diff
@@ -0,0 +1,69 @@
+     *
+     * @return A string representing the task type. Default is "Normal task".
+     */
+    public String taskType() {
```
Can I ask is this method is used? If it is meant to be overridden, maybe make it an abstract method?

Jianwen0451 commented on src/main/java/wenjigglybot/InvalidCommandException.java:
```diff
@@ -0,0 +1,9 @@
+
+public class InvalidCommandException extends Exception {
+    @Override
+    public String toString() {
```
Maybe it is better to make use of getMessage() method instead of override toString() method?

wenjebs committed:
Add assertions to WenJigglyBot class

wenjebs committed:
Add assertions to validate input in Parser class

Currently, the Parser class lacks input validation, which could lead
to unpredictable behavior or exceptions during runtime when handling
incorrectly formatted input data.

This needs to change because there is no clear mechanism to catch
improper inputs early, leading to potential debugging difficulties.

Add assertions in methods like processEventTask, processDeadlineTask,
parseCommand, and parseTask to verify the integrity of input data before
proceeding with processing. This helps detect incorrect input formats at runtime.

Assertions were chosen because they are an effective, lightweight way to
ensure input validation during development and testing, without
introducing runtime overhead in production environments. They also help
developers identify issues faster during debugging.

wenjebs committed:
Refactor parser class

wenjebs committed:
Refactor WenJigglyBot class

wenjebs committed:
Add help command

wenjebs committed:
Add help functionality

wenjebs committed:
Merge pull request #2 from wenjebs/A-Assertions

A assertions

wenjebs committed:
Resolve merge conflict by incorporating both suggestions

wenjebs committed:
Merge pull request #3 from wenjebs/A-CodeQuality

A code quality

wenjebs committed:
Merge pull request #4 from wenjebs/BCD-Extension

Bcd extension

wenjebs committed:
Change image

wenjebs committed:
Update readme

wenjebs committed:
Finalise readme

wenjebs committed:
Add more javadocs

wenjebs committed:
Fix file directories
